File: src/app/Models/VerificationDetail.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class VerificationDetail extends Model
{
    protected $fillable = ['user_id', 'ktp_image', 'ktp_selfie', 'status', 'reviewed_at'];

    protected function casts(): array
    {
        return ['reviewed_at' => 'datetime'];
    }

    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class);
    }
}

File: backend/internal/model/verification_detail.go
package model

import (
	"time"

	"github.com/google/uuid"
)

type VerificationDetail struct {
	ID         uuid.UUID  `gorm:"type:uuid;primaryKey;default:gen_random_uuid()" json:"id"`
	UserID     uuid.UUID  `gorm:"type:uuid;not null;uniqueIndex" json:"user_id"`
	KtpImage   string     `gorm:"size:255" json:"ktp_image"`
	KtpSelfie  string     `gorm:"size:255" json:"ktp_selfie"`
	Status     string     `gorm:"size:20;default:'pending'" json:"status"` // pending, approved, rejected
	ReviewedAt *time.Time `json:"reviewed_at"`
	CreatedAt  time.Time  `json:"created_at"`
	UpdatedAt  time.Time  `json:"updated_at"`

	User User `gorm:"foreignKey:UserID;constraint:OnDelete:CASCADE" json:"user,omitempty"`
}
